src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/FindAllReferencesHandlerTest.cs

```diff
@@ -153,6 +159,7 @@ public async Task HandleRequestAsync_HtmlProjection_InvokesHtmlLanguageServer()
                     a => AssertVSReferenceItem(expectedLocation1, a),
                     b => AssertVSReferenceItem(expectedLocation2, b));
                 progressReported = true;
+                completedTokenSource.CancelAfter(0);
```

Comment:
What's the difference between this and an immediate cancellation?

Reply:
`Cancel()` waits for the callbacks registered to the `CancellationToken` to complete before returning. `CancelAfter(0)` schedules immediate cancellation but doesn't do any of the work synchronously.